src/frontend/Pretty_printing.ml

```diff
@@ -385,21 +395,45 @@ let pp_opt_block ppf block_name opt_block =
   Fmt.option ~none:Fmt.nop (pp_block block_name) ppf opt_block
 
 let pp_program ppf
-    { functionblock= bf
-    ; datablock= bd
-    ; transformeddatablock= btd
-    ; parametersblock= bp
-    ; transformedparametersblock= btp
-    ; modelblock= bm
-    ; generatedquantitiesblock= bgq } =
+    ({ comments0
+     ; functionblock= bf
+     ; comments1
+     ; datablock= bd
+     ; comments2
+     ; transformeddatablock= btd
+     ; comments3
+     ; parametersblock= bp
+     ; comments4
+     ; transformedparametersblock= btp
+     ; comments5
+     ; modelblock= bm
+     ; comments6
+     ; generatedquantitiesblock= bgq
+     ; comments7 } :
+      untyped_program) =
   Format.pp_open_vbox ppf 0 ;
+  Fmt.list pp_comment ppf comments0 ;
+  if not (List.is_empty comments0) then Fmt.cut ppf () ;
```

Comment:
Maybe we can pull out these two lines into pp_comments for clarity/clutter reduction?